crates/store/src/db/sql.rs

```diff
@@ -184,41 +189,125 @@ pub fn select_accounts_by_ids(
     Ok(result)
 }
 
-/// Select account deltas by account id and block range from the DB using the given [Connection].
+/// Selects and merges account deltas by account id and block range from the DB using the given
+/// [Connection].
 ///
 /// # Note:
 ///
 /// `block_start` is exclusive and `block_end` is inclusive.
 ///
 /// # Returns
 ///
-/// The account deltas, or an error.
-pub fn select_account_deltas(
+/// The resulting account delta, or an error.
+pub fn select_account_delta(
     conn: &mut Connection,
     account_id: AccountId,
     block_start: BlockNumber,
     block_end: BlockNumber,
-) -> Result<Vec<AccountDelta>> {
-    let mut stmt = conn.prepare_cached(
+) -> Result<Option<AccountDelta>> {
+    let mut select_nonce_stmt = conn.prepare_cached(
         "
         SELECT
-            delta
+            nonce
         FROM
             account_deltas
         WHERE
             account_id = ?1 AND block_num > ?2 AND block_num <= ?3
         ORDER BY
-            block_num ASC
+            block_num DESC
+        LIMIT 1
     ",
     )?;
 
-    let mut rows = stmt.query(params![u64_to_value(account_id), block_start, block_end])?;
-    let mut result = Vec::new();
+    let account_id = u64_to_value(account_id);
+    let nonce = match select_nonce_stmt
+        .query_row(params![account_id, block_start, block_end], |row| row.get::<_, u64>(0))
+    {
+        Ok(nonce) => nonce.try_into().map_err(DatabaseError::InvalidFelt)?,
+        Err(rusqlite::Error::QueryReturnedNoRows) => return Ok(None),
+        Err(e) => return Err(e.into()),
+    };
+
+    #[derive(num_derive::FromPrimitive)]
+    enum RecordType {
+        StorageScalars = 0,
+        StorageMapValues,
+        FungibleAssets,
+        NonFungibleAssets,
+    }
+    let mut select_merged_deltas_stmt =
+        conn.prepare_cached(include_str!("sql-queries/select_merged_deltas.sql"))?;
```

Comment:
Do we know if this `UNION ALL` approach is better than just having four separate queries?

How do the `explain query plan` compare?

Reply:
We can expect just very little benefit in performance here, but I haven't done any benchmarks yet.